expression/builtin_time_vec.go

```diff
@@ -2237,11 +2237,56 @@ func (b *builtinTimestampDiffSig) vecEvalInt(input *chunk.Chunk, result *chunk.C
 }
 
 func (b *builtinUnixTimestampIntSig) vectorized() bool {
-	return false
+	return true
 }
 
+// evalInt evals a UNIX_TIMESTAMP(time).
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_unix-timestamp
 func (b *builtinUnixTimestampIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil && terror.ErrorEqual(types.ErrWrongValue.GenWithStackByArgs(types.TimeStr, buf), err) {
```

Comment:
You've missed the situation that `err != nil` but not `types.ErrWrongValue`?

Reply:
I found that the non-vectorized implement also don't handle the `err != nil` but not `types.ErrWrongValue` situation.

Reply:
Because it will hit `isNull` in the non-vectorized implemention.
```go
	if isNull {
		return 0, true, nil
	}
```